Apps/Examples/Examples/All Examples/OfflineManagerExample.swift

```diff
@@ -0,0 +1,422 @@
+        }
+
+        TileStore.getInstance().allTileRegions { result in
+            self.logDownloadResult(message: "Tile regions:", result: result)
+        }
+        Log.info(forMessage: "\n", category: "Example")
+    }
+
+    // Remove downloaded region and style pack
+    private func removeTileRegionAndStylePack(completion: (() -> Void)? = nil ) {
+        // Clean up after the example. Typically, you'll have custom business
+        // logic to decide when to evict tile regions and style packs
+
+        // Remove the tile region with the tile region ID.
+        // Note this will not remove the downloaded tile packs, instead, it will
+        // just mark the tileset as not a part of a tile region. The tiles still
+        // exists in a predictive cache in the TileStore.
+        TileStore.getInstance().removeTileRegion(forId: tileRegionId)
+
+        // Set the disk quota to zero, so that tile regions are fully evicted
+        // when removed. The TileStore is also used when `ResourceOptions.isLoadTilePacksFromNetwork`
+        // is `true`, and also by the Navigation SDK.
+        // This removes the tiles from the predictive cache.
+        TileStore.getInstance().setOptionForKey(TileStoreOptions.diskQuota, value: 0)
+
+        // Remove the style pack with the style uri.
+        // Note this will not remove the downloaded style pack, instead, it will
+        // just mark the resources as not a part of the existing style pack. The
+        // resources still exists in the ambient cache.
+        offlineManager.removeStylePack(for: .outdoors)
+
+        // Remove the existing style resources from ambient cache using cache manager.
+        let cacheManager = CacheManager(options: mapInitOptions.resourceOptions)
+        cacheManager.clearAmbientCache { _ in
+            completion?()
+        }
+    }
+
+    // MARK: - State changes
+
+    @IBAction func didTapButton(_ button: UIButton) {
+        switch state {
+        case .unknown:
+            state = .initial
+        case .initial:
+            downloadTileRegions()
+        case .downloading:
+            // Cancel
+            cancelDownloads()
+        case .downloaded:
+            state = .mapViewDisplayed
+        case .mapViewDisplayed:
+            showDownloadedRegions()
+            state = .finished
+        case .finished:
+            removeTileRegionAndStylePack { [weak self] in
+                self?.showDownloadedRegions()
+                self?.state = .initial
+            }
+        }
+    }
+
+    private var state: State = .unknown {
+        didSet {
+            Log.warning(forMessage: "Changing state from \(oldValue) -> \(state)", category: "Debug")
+
+            switch (oldValue, state) {
+            case (_, .initial):
+                resetUI()
+                Log.warning(forMessage: "Enabling HTTP stack network connection", category: "Example")
+                NetworkConnectivity.getInstance().setMapboxStackConnectedForConnected(true)
```

Comment:
I think we should set the network connectivity to true before the activity exits, otherwise I think other MapViews in the app will stop working..